src/main/java/reactor/core/publisher/FluxPeekFuseable.java

```diff
@@ -258,7 +263,7 @@ public T poll() {
 					throw Exceptions.propagate(Operators.onOperatorError(s, e, v));
 				}
 			}
-			if (v == null && sourceMode == SYNC) {
+			if (v == null && (d || sourceMode == SYNC)) {
```

Comment:
Doesn't work, you have to read done then q.isEmpty. Reversed and you may see an empty queue but when you see done after that, the queue may have received more values from upstream. Remember that poll may run async to onNext/onComplete.